azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java

```diff
@@ -505,6 +501,17 @@ public void run() {
     }
   }
 
+  @Override
+  protected void finalizeFlow(final ExecutableFlow flow, final String reason,
+      @Nullable final Throwable originalError, final Status finalFlowStatus) {
+    super.finalizeFlow(flow, reason, originalError, finalFlowStatus);
+    try {
+      this.containerizedImpl.deleteContainer(flow.getExecutionId());
+    } catch (ExecutorManagerException e) {
+      logger.info("Failed to delete container when finalizing flow", e);
```

Comment:
can we add exec_id here. It might help to get more logs associated with the execution for debug.
Also should the log be marked as ERROR?

Reply:
It seems WARN is more appropriate b/c maybe the container is not up indeed.